### ISSUE TITLE
Pacaur detects phantom unsatisfiable dependencies

### ISSUE THREAD

**Issue by gandalf3** [rmarquis/pacaur#668]
###### Version

`pacaur -v`

```
4.7.5.6.g6803b7b
```

###### Description

pacaur claims to be unable to find these five packages, and also seems to believe all aur packages depend on them:
```
python2-qscintilla-qt4<2.10
python2-sip<4.19.1
python-qscintilla-qt4-common<2.10
qscintilla-qt4<2.10
sip<4.19.1
```
However, they are all installed and up-to-date:   
`pacman -Q python2-qscintilla-qt4 python2-sip python-qscintilla-qt4-common qscintilla-qt4 sip`

```
python2-qscintilla-qt4 2.10-1
python2-sip 4.19.1-1
python-qscintilla-qt4-common 2.10-1
qscintilla-qt4 2.10-1
sip 4.19.1-1
```
If I update the dependent package (e.g. `unity-editor`), then the next outdated aur package will take its place.

###### Output

`pacaur -Syua`

```
:: Synchronizing package databases...
 core is up to date
 extra is up to date
 community is up to date
 multilib is up to date
 archlinuxgr is up to date
:: Starting AUR upgrade...
# long list of packages not present in the aur, omitted for brevity (none of the packages in the error messages are mentioned)
# list of packages in IgnorePkg (same as above)
:: resolving dependencies...
:: no results found for python2-qscintilla-qt4<2.10 (dependency tree: unity-editor python2-qscintilla-qt4)
:: no results found for python2-sip<4.19.1 (dependency tree: unity-editor python2-sip)
:: no results found for python-qscintilla-qt4-common<2.10 (dependency tree: unity-editor python-qscintilla-qt4-common)
:: no results found for qscintilla-qt4<2.10 (dependency tree: unity-editor qscintilla-qt4)
:: no results found for sip<4.19.1 (dependency tree: unity-editor sip)
```

###### Debug output

`bash -x pacaur -Syua`

Output too long to fit, instead uploaded here: [pacaur-errors.txt](https://github.com/rmarquis/pacaur/files/852650/pacaur-errors.txt)

**Comment by rmarquis** [comment]
All of the 5 missing dependencies belongs to the [qgis](https://aur.archlinux.org/packages/qgis/) package. Seems the reverse dependency tree solver of the errors found isn't robust enough yet. This might be related to versioned dependencies.

**Comment by rmarquis** [comment]
Fixed in 1c116b15b170efdad97304a1c1a5397b67b451a6.
